library/src/main/java/com/google/android/material/motion/runtime/Scheduler.java

```diff
@@ -172,6 +172,38 @@ public void addPlan(Plan plan, Object target) {
     getTargetScope(target).commitPlan(planInfo);
   }
 
+  /**
+   * Adds a {@link NamedPlan} to this scheduler.
+   * @param plan the {@link NamedPlan} to add to the scheduler.
+   * @param name the name by which this plan can be identified.
+   * @param target the target on which the plan will operate.
+   */
+  public void addNamedPlan(NamedPlan plan, String name, Object target) {
+    if (name == null || name.length() == 0) {
+      throw new AssertionError("A NamedPlan must have a name with more than zero characters");
```

Comment:
I was looking for something similar to `NSParameterAssert` in Java and settled on `AssertionError` - I think `IllegalArgumentException` gives a better indication to the client about what went wrong though. Thanks!